bpf/lib/nodeport.h

```diff
@@ -327,20 +327,41 @@ static __always_inline int find_dsr_v6(struct __ctx_buff *ctx, __u8 nexthdr,
 
 static __always_inline int
 nodeport_extract_dsr_v6(struct __ctx_buff *ctx, struct ipv6hdr *ip6,
+			struct ipv6_ct_tuple *tuple, int l4_off,
 			union v6addr *addr, __be16 *port, bool *dsr)
 {
 	struct dsr_opt_v6 opt __align_stack_8 = {};
+	struct ipv6_ct_tuple tmp = *tuple;
 	int ret;
 
+	if (tuple->nexthdr == IPPROTO_TCP) {
+		union tcp_flags tcp_flags = {};
+
+		if (l4_load_tcp_flags(ctx, l4_off, &tcp_flags) < 0)
+			return DROP_CT_INVALID_HDR;
+
+		ipv6_ct_tuple_reverse(&tmp);
+
+		if (!(tcp_flags.value & TCP_FLAG_SYN)) {
+			*dsr = ct_has_dsr_egress_entry6(get_ct_map6(&tmp), &tmp);
+			*port = 0;
+			return 0;
+		}
+	}
```

Comment:
Could we have a comment for this whole block, to explain why it's needed? IMO, what you have in the commit description should be good enough.

Reply:
I would add a bit of details to help the reader go from `TCP connections have the DSR Option only in their SYN packet` to "we need to parse the TCP packets". Right now, *why* that means we need this parsing block isn't super obvious.

Reply:
Expanded the comment in `nodeport_extract_dsr_v4()`.